libOTe/Tools/SubfieldPprf.h

```diff
@@ -0,0 +1,1437 @@
+                    }
+                }
+                // pprf.setTimePoint("SilentMultiPprfSender.alloc " + std::to_string(treeIdx));
+
+                // This thread will process 8 trees at a time. It will interlace
+                // the sets of trees are processed with the other threads. 
+                {
+                    // The number of real trees for this iteration.
+                    min = std::min<u64>(8, pprf.mPntCount - treeIdx);
+                    //gTimer.setTimePoint("send.start" + std::to_string(treeIdx));
+
+                    // Populate the zeroth level of the GGM tree with random seeds.
+                    prng.get(getLevel(0, treeIdx));
+
+                    // Allocate space for our sums of each level.
+                    sums[0].resize(pprf.mDepth);
+                    sums[1].resize(pprf.mDepth);
+
+                    // For each level perform the following.
+                    for (u64 d = 0; d < pprf.mDepth; ++d)
+                    {
+                        // The previous level of the GGM tree.
+                        auto level0 = getLevel(d, treeIdx);
+
+                        // The next level of theGGM tree that we are populating.
+                        auto level1 = getLevel(d + 1, treeIdx);
+
+                        // The total number of children in this level.
+                        auto width = static_cast<u64>(level1.size());
+
+                        // For each child, populate the child by expanding the parent.
+                        for (u64 childIdx = 0; childIdx < width; )
+                        {
+                            // Index of the parent in the previous level.
+                            auto parentIdx = childIdx >> 1;
+
+                            // The value of the parent.
+                            auto& parent = level0[parentIdx];
+
+                            // The bit that indicates if we are on the left child (0)
+                            // or on the right child (1).
+                            for (u64 keep = 0; keep < 2; ++keep, ++childIdx)
+                            {
+                                // The child that we will write in this iteration.
+                                auto& child = level1[childIdx];
+
+                                // The sum that this child node belongs to.
+                                auto& sum = sums[keep][d];
+
+                                // Each parent is expanded into the left and right children 
+                                // using a different AES fixed-key. Therefore our OWF is:
+                                //
+                                //    H(x) = (AES(k0, x) + x) || (AES(k1, x) + x);
+                                //
+                                // where each half defines one of the children.
+                                aes[keep].hashBlocks<8>(parent.data(), child.data());
+
+                                if (d < pprf.mDepth - 1) {
```

Comment:
a nested if statement like this might have an impact on performance. Could be work checking